source/io/module_eggShell.js

```diff
@@ -372,7 +372,9 @@
             return result;
         };
 
-        Module.eggShell.executeSlices = publicAPI.eggShell.executeSlices = function (slices) {
+        Module.eggShell.executeSlicesUntilWait = publicAPI.eggShell.executeSlicesUntilWait = function (slices) {
+            // returns -1 if should be called again ASAP, 0 if nothing to run, or positive value N if okay
```

Comment:
Should the comment be "returns < 0 if should be called again ASAP"?

Reply:
Changed.

Reply:
Ugh, I hate how I can't reply directly to your non-review comment

> What if in ExecuteSlices we instead keep track of the last returned returned ExecutionState in the user shell along with a timestamp? With that information ExecuteSlices can determine if it is being called too frequently and print a warning.

> This would be helpful in general for Vireo integrators to be warned that they can pump the runtime more efficiently.

> It would probably need a check to see if the last value was invalidated, ie a Set_Occurence was called by the HTTP nodes between calls to ExecuteSlices so a warning is not necessary.

I like this idea but I don't think I want to try to do it in this PR.  I think I would make the check only if the run loop didn't have anything to do for multiple calls in a row, and the previous return value was > 0.  (A call to Set_Ocurrence would schedule something, so the run loop would run and reset the counter.)